opentelemetry-api/src/opentelemetry/attributes/__init__.py

```diff
@@ -108,3 +111,73 @@ def _create_immutable_attributes(
     attributes: types.Attributes,
 ) -> types.Attributes:
     return MappingProxyType(attributes.copy() if attributes else {})
+
+
+_DEFAULT_LIMIT = 128
+
+
+class BoundedDict(MutableMapping):
+    """An ordered dict with a fixed max capacity.
+
+    Oldest elements are dropped when the dict is full and a new element is
+    added.
+    """
+
+    def __init__(
+        self,
+        maxlen: Optional[int] = _DEFAULT_LIMIT,
+        attributes: types.Attributes = None,
```

Comment:
`attributes` makes sense in terms of what we are using `BoundedDict` for, but for the class itself it doesn't make much sense. Should we have a different name instead of `BoundedDict`?

Reply:
It would be good to make a separation from the SDK `BoundedDict` too.

Reply:
Yeah I considered for a minute making `Attributes` a class to replace `BoundedDict`, wasn't sure if this would be doable without breaking Attributes' current interface though

Reply:
Maybe just `BoundedAttributes` or something?

Reply:
Done.